lib/TPP/CMakeLists.txt

```diff
@@ -19,6 +19,21 @@ add_mlir_library(TPPPipeline
 
   LINK_LIBS PUBLIC
     MLIRIR
+    MLIRMeshDialect
```

Comment:
That's a whole lotta libraries we don't use. Why add them here?

Reply:
it's from registerAllDialects in DefaultPipeline.cpp
Without this many linking errors generated for FAILED: bin/torch-mlir-opt
Like this: 
```
GpuPipeline.cpp:(.text._ZN4mlir15DialectRegistry6insertINS_4math11MathDialectENS_6memref13MemRefDialectEJNS_4mesh11MeshDialectENS_10ml_program16MLProgramDialectENS_3mpi10MPIDialectENS_5nvgpu12NVGPUDialectENS_4NVVM11NVVMDialectENS_3omp13OpenMPDialectENS_3pdl10PDLDialectENS_10pdl_interp16PDLInterpDialectENS_5quant19QuantizationDialectENS_5ROCDL12ROCDLDialectENS_3scf10SCFDialectENS_5shape12ShapeDialectENS_13sparse_tensor19SparseTensorDialectENS_5spirv12SPIRVDialectENS_6tensor13TensorDialectENS_4tosa11TosaDialectENS_9transform16TransformDialectENS_2ub9UBDialectENS_6vector13VectorDialectENS_9x86vector16X86VectorDialectENS_5xegpu12XeGPUDialectEEEEvv[_ZN4mlir15DialectRegistry6insertINS_4math11MathDialectENS_6memref13MemRefDialectEJNS_4mesh11MeshDialectENS_10ml_program16MLProgramDialectENS_3mpi10MPIDialectENS_5nvgpu12NVGPUDialectENS_4NVVM11NVVMDialectENS_3omp13OpenMPDialectENS_3pdl10PDLDialectENS_10pdl_interp16PDLInterpDialectENS_5quant19QuantizationDialectENS_5ROCDL12ROCDLDialectENS_3scf10SCFDialectENS_5shape12ShapeDialectENS_13sparse_tensor19SparseTensorDialectENS_5spirv12SPIRVDialectENS_6tensor13TensorDialectENS_4tosa11TosaDialectENS_9transform16TransformDialectENS_2ub9UBDialectENS_6vector13VectorDialectENS_9x86vector16X86VectorDialectENS_5xegpu12XeGPUDialectEEEEvv]+0x1b9): undefined reference to `mlir::detail::TypeIDResolver<mlir::mpi::MPIDialect, void>::id'
```
or in shorten:
```
GpuPipeline.cpp: undefined reference to `mlir::detail::TypeIDResolver<mlir::mpi::MPIDialect, void>::id'
```

So this target uses these libraries, I don't know why in standalone mode this problem doesn't occur.
I am not sure how to fix them it in different way.

Reply:
I found a slightly better solution and used ${mlir_dialect_libs}. I don't know why I only noticed this variable now.